### ISSUE TITLE
Invalid type for RHS when using Mat3#multiply! or Mat4#multiply!

### ISSUE THREAD

**Issue by flashguardian13** [nilium/ruby-snowmath#2]
I will let this bug speak for itself.

```
irb(main):001:0> require 'snow-math'
=> true
irb(main):002:0> snowmat = Snow::Mat3[ 0, 0.5, 100,   -2, 0, 50,   0, 0, 1 ]
=> <Snow::Mat3:0x2bcfb30 0.0 0.5 100.0 -2.0 0.0 50.0 0.0 0.0 1.0>
irb(main):003:0> snowmat * snowmat
=> <Snow::Mat3:0x2c95dc8 -1.0 0.0 125.0 0.0 -1.0 -150.0 0.0 0.0 1.0>
irb(main):004:0> snowmat.multiply(snowmat)
=> <Snow::Mat3:0x2df17f0 -1.0 0.0 125.0 0.0 -1.0 -150.0 0.0 0.0 1.0>
irb(main):005:0> snowmat.multiply!(snowmat)
NameError: uninitialized constant Snow::Mat3::Mat3
Did you mean?  Snow::Mat3
               Snow::Mat3Array
               Snow::Mat4
               Math
        from C:/Ruby23/lib/ruby/gems/2.3.0/gems/snow-math-1.7.1/lib/snow-math/mat3.rb:124:in `multiply!'
        from (irb):5
        from C:/Ruby23/bin/irb.cmd:19:in `<main>'
```

The problem seems to originate in `Mat3#multiply!`. The solution seems to be to start from the main scope when type-checking in the `case/when` block, as was done with the multiply method.
```
  def multiply!(rhs)
    multiply rhs, case rhs
      when ::Snow::Mat3, Numeric then self
      when ::Snow::Vec3 then rhs
      else raise TypeError, "Invalid type for RHS"
      end
  end
```

**Comment by nilium** [comment]
Seems like Mat4 also has this bug. Would you like to submit a patch for this?

**Comment by flashguardian13** [comment]
Sure. I'll probably drop something here when I have a moment.

**Comment by flashguardian13** [comment]
I have a patch ready to push, but I don't have push permissions to the repository.

**Comment by nilium** [comment]
You can fork it, push to your fork, and submit a pull request from there. These might help:

- https://help.github.com/articles/creating-a-pull-request/
- https://help.github.com/articles/creating-a-pull-request-from-a-fork/

**Comment by flashguardian13** [comment]
Here you are! Let me know if you want the version number bumped, too.

https://github.com/nilium/ruby-snowmath/pull/3

**Comment by nilium** [comment]
Thanks. I'll look at getting it merged later today when I'm off work. Don't worry about bumping the version — I'll take care of that.

**Comment by nilium** [comment]
This is fixed in 1.7.2 onward. 1.7.3 was pushed to fix the SPDX license identifier as well (since I missed that before tagging 1.7.2).
